student.rb

```diff
@@ -1,12 +1,20 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(id, age, classroom, name = 'Unknown', parent_permission: true)
+  attr_reader :classroom
+
+  def initialize(id, age, classroom = nil, name = 'Unknown', parent_permission: true)
     @classroom = classroom
+    add_student_classroom if clasroom
     super(id, age, name, parent_permission)
   end
 
   def play_hooky
     '¯\\(ツ)/¯'
   end
+
+  def add_student_classroom(classroom)
+    @classroom = classroom
+    clasroom.add_student(self) unless classroom.students.include?(self)
+  end
```

Comment:
- [ ] To make your code more clean, you may consider not using the methods of other classes in a different class as you did here. because if in the future the `add_student` method is modified, you will have a lot of kind of bugs in your program.
Hint: To correct that you may consider saying:
```rb
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```